frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -200,6 +198,10 @@ export type WatchK8sResources<R extends ResourcesObject> = {
   [k in keyof R]: WatchK8sResource;
 };
 
+export type WatchK8sResourcesOptional<R extends ResourcesObject> = {
+  [k in keyof R]: WatchK8sResource & { optional?: boolean };
```

Comment:
Sorry, I'm a little unclear on the need for this. If `optional` doesn't change the behavior, is this only to "document" that it's optional in code? Would a comment work just a well, then we don't need the special type?

Reply:
`optional` does not change the behavior of the k8s watch hook but some extensions use the prop to determine how to handle the fetch errors for example (ie if the resource does not exist and optional is true, than the resource is ignored, if optional is false it would show some error)